providers-auth.md

```diff
@@ -0,0 +1,190 @@
+## Configurations managed with a service principal
```

Comment:
Can we alphabetically order the providers? Also might be usefull to add their namespace. For example MSGraph:Groups to make it easier to build filters based on this document. @brad-Simeon @Lance-Simeon